pkg/controller/cloneset/update/cloneset_update.go

```diff
@@ -143,6 +143,10 @@ func calculateUpdateCount(coreControl clonesetcore.Control, strategy appsv1alpha
 
 	maxUnavailable, _ := intstrutil.GetValueFromIntOrPercent(
 		intstrutil.ValueOrDefault(strategy.MaxUnavailable, intstrutil.FromString(appsv1alpha1.DefaultCloneSetMaxUnavailable)), totalReplicas, true)
+	if strategy.MaxSurge != nil {
```

Comment:
How about this

```golang
roundUp := strategy.MaxSurge == nil
maxUnavailable, _ := intstrutil.GetValueFromIntOrPercent(
    intstrutil.ValueOrDefault(strategy.MaxUnavailable, intstrutil.FromString(appsv1alpha1.DefaultCloneSetMaxUnavailable)), totalReplicas, roundUp)
```

Reply:
done, the ci check was failed by timeout, how to recheck it ?